catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -1051,6 +1170,41 @@ private Metacard saveMetacard(Metacard metacard)
     return catalogFramework.create(new CreateRequestImpl(metacard)).getCreatedMetacards().get(0);
   }
 
+  private List<Metacard> getWorkspaceQueryMetacards(Map<String, Object> workspace) {
+    return ((List<Map<String, Object>>) workspace.get("queries"))
+        .stream()
+        .map(queryTransformer::transformJsonIntoMetacard)
+        .collect(Collectors.toList());
+  }
+
+  private WorkspaceMetacardImpl getWorkspaceMetacard(String id) {
+    return util.getMetacards(Collections.singletonList(id), WorkspaceConstants.WORKSPACE_TAG)
+        .values()
+        .stream()
+        .map(Result::getMetacard)
+        .map(WorkspaceMetacardImpl::new)
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new NotFoundException(
+                    String.format("Could not find workspace metacard with id %s", id)));
+  }
+
+  private List<String> getWorkspaceAssociations(Metacard workspace, String association) {
+    Attribute attr = workspace.getAttribute(association);
+    return attr == null
```

Comment:
:pencil2: these two lines could be combined

Reply:
I prefer to keep it this way to reduce nesting